### PR TITLE
Branch

### PR DESCRIPTION
**Pull request by ghost**
WTF I'M DOING?????

### CODE REVIEW

File: Web/Controllers/Investment/StockController.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+using DataAccess;
+using DataAccess.Models;
+using Microsoft.AspNetCore.Mvc;
+using Web.ViewModels;
+
+namespace Web.Controllers.Investment
+{
+    public class StockController : Controller
+    {
+        private readonly Repository _repository;
```
Comment (t-eliks) [inline Web/Controllers/Investment/StockController.cs]: Please keep consistent code style: private variables start with lowercase letters, public with Uppercase. No underscores.

File: Web/Controllers/Investment/StockController.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        
+        # region Get methods
+
+        [HttpGet]
+        public IActionResult OpenStocksReport()
+        {
+            var userStocks = GetUserStocks();
+            var stocks = GetStockValuesAndPrognosis(userStocks).Result;
+            return View("StocksReport", stocks);
+        }
```
Comment (t-eliks) [inline Web/Controllers/Investment/StockController.cs:38]: Please add spacing between methods. And an empty line before a return statement.

Comment (t-eliks) [reply Web/Controllers/Investment/StockController.cs:38]: Refer to other controllers if need arises.

File: Web/Controllers/Investment/StockController.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        # endregion
+        
+        # region Post methods
+        [HttpPost]
+        public IActionResult CreateNewStock(StockViewModel stockViewModel)
+        {
+            if (ValidateData())
+            {
+                TempData["Error"] = "Neteisingai įvesti duomenys";
```
Comment (t-eliks) [inline Web/Controllers/Investment/StockController.cs]: Please keep consistency within project and not use model state annotations. Refer to other controllers for implementation details.

File: Web/Controllers/Investment/StockController.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        [HttpPost]
+        public IActionResult UpdateStock(StockViewModel stock)
+        {
+            if (ValidateData())
+            {
+                TempData["Error"] = "Neteisingai įvesti duomenys";
+                return View("StockForm", stock);
+            }
+            TempData["Success"] = "Akcija sėkmingai atnaujinta";
+            UpdateStock(stock, stock.Id);
```
Comment (t-eliks) [inline Web/Controllers/Investment/StockController.cs]: What's the point of passing the id as a separate parameter when it always comes with the viewmodel anyway?

File: Web/ViewModels/StockReportViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        /*
+         * Won't map to decimal, because Json values is strings
+         * and don't use setter to set value, using Convert,
+         * from API floating point values comes with '.' not ','
+         * fails to convert
+         */
+        public DateTime datetime { get; set; }
+        
+        public string open { get; set; }
```
Comment (t-eliks) [inline Web/ViewModels/StockReportViewModel.cs:20]: Lowercase public properties

File: Web/Views/Stock/StocksReport.cshtml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+@{
+    ViewData["Title"] = "Akcijų ataskaita";
+    ViewData["Menu"] = "StockReport";
+    ViewData["BreadCrumbs"] = new BreadCrumb()
+        .Append(SiteCrumbs.Home(Url))
+        .Append(SiteCrumbs.StockReport(Url));
+    Layout = "~/Views/Shared/_Layout.cshtml";
+}
+<script type="text/javascript">
```
Comment (t-eliks) [inline Web/Views/Stock/StocksReport.cshtml]: The script section should be at the bottom of the file, under @section Scripts { `code here` }

Comment (t-eliks) [reply Web/Views/Stock/StocksReport.cshtml]: Refer to PlanPage.cshtml

**Review by t-eliks — CHANGES_REQUESTED** [review]

File: Web/ViewModels/StockReportViewModel.cs (hunk 5 above)
Comment (ghost) [reply Web/ViewModels/StockReportViewModel.cs:20]: Used for json deserialize, names should match json objects with C# object

**Pull request closed without merging**